storage/shard.go

```diff
@@ -1481,10 +1492,22 @@ func (s *dbShard) Bootstrap(
 	// Now iterate flushed time ranges to determine which blocks are
 	// retrievable before servicing reads
 	fsOpts := s.opts.CommitLogOptions().FilesystemOptions()
-	entries := fs.ReadInfoFiles(fsOpts.FilePathPrefix(), s.namespace.ID(), s.shard,
+	readInfoFilesResults, err := fs.ReadInfoFiles(fsOpts.FilePathPrefix(), s.namespace.ID(), s.shard,
 		fsOpts.InfoReaderBufferSize(), fsOpts.DecodingOptions())
-	for _, info := range entries {
-		at := xtime.FromNanoseconds(info.Start)
+	if err != nil {
```

Comment:
This is what I meant with the comment about the usage of `ReadInfoFiles`. Most people will look at this and think it's a bug it doesn't terminate early in the if block.

Reply:
ok fair enough, I'll just make it return a slice of results and callers will have to check each error individually

Reply:
done